genesis/utils/usd.py

```diff
@@ -0,0 +1,312 @@
+                                mdl_material_name = mdl_sub_identifier_attr.Get()
+                                diffuse_texture_path, normal_texture_path, diffuse_tint, orm_texture_path = (
+                                    extract_texture_paths_from_mdl(mdl_asset_path)
+                                )
+
+                                if diffuse_tint:
+                                    diffuse_tint = np.array(list(diffuse_tint) + [1.0], dtype=float)  # RGB + A
+                                else:
+                                    diffuse_tint = np.ones(4, dtype=float)
+
+                                color_texture = None
+                                normal_texture = None
+                                opacity_texture = None
+                                roughness_texture = None
+                                metallic_texture = None
+
+                                if diffuse_texture_path:
+                                    diffuse_texture_path = os.path.normpath(
+                                        os.path.join(os.path.dirname(mdl_asset_path), diffuse_texture_path)
+                                    )
+                                    diffuse_image = mu.PIL_to_array(Image.open(diffuse_texture_path))
+                                    color_texture = mu.create_texture(diffuse_image, diffuse_tint, "srgb")
+                                    gs.logger.info(f"Get Diffuse texture: {diffuse_texture_path}")
+
+                                if normal_texture_path:
+                                    normal_texture_path = os.path.normpath(
+                                        os.path.join(os.path.dirname(mdl_asset_path), normal_texture_path)
+                                    )
+                                    normal_image = mu.PIL_to_array(Image.open(normal_texture_path))
+                                    normal_texture = mu.create_texture(normal_image, None, "linear")
+                                    gs.logger.info(f"Get Normal texture: {normal_texture_path}")
+
+                                if orm_texture_path:
+                                    orm_texture_path = os.path.normpath(
+                                        os.path.join(os.path.dirname(mdl_asset_path), orm_texture_path)
+                                    )
+                                    orm_image = mu.PIL_to_array(Image.open(orm_texture_path))
+                                    # Split ORM image into separate textures
+                                    opacity_texture = mu.create_texture(orm_image[:, :, 0], None, "linear")
+                                    roughness_texture = mu.create_texture(orm_image[:, :, 1], None, "linear")
+                                    metallic_texture = mu.create_texture(orm_image[:, :, 2], None, "linear")
+                                    gs.logger.info(f"Get ORM texture: {orm_texture_path}")
+
+                                mesh_surface.update_texture(
```

Comment:
You should be able to merge these 2 branches in one. Too many code duplications here. I mean, share init to None and mesh surface instantiation.

Reply:
I can share init to None but sharing `mesh_surface.update_texture` is not quite elegant